fast64_internal/sm64/sm64_objects.py

```diff
@@ -1794,6 +1799,126 @@ def get_behavior_params(self):
         return self.bparams
 
 
+def check_parents_has_cond(obj: bpy.types.Object, cond_cb: Callable[[bpy.types.Object], bool]):
+    if not cond_cb(obj):
+        if obj.parent:
+            return check_parents_has_cond(obj.parent, cond_cb)
+        return False
+    return True
+
+
+def check_children_for_cond(obj: bpy.types.Object, cond_cb: Callable[[bpy.types.Object], bool]):
+    if cond_cb(obj):
+        return True
+    if len(obj.children):
+        for c in obj.children:
+            if check_children_for_cond(c, cond_cb):
+                return True
```

Comment:
```suggestion
    for c in obj.children:
        if check_children_for_cond(c, cond_cb):
            return True
```

Reply:
if obj.children is empty it still allows itself to be iterated, just ends up exiting immediately

Reply:
yeah, indeed? the len check is pointless

Reply:
yee id rather leave it out then

Reply:
then remove it :angryjoy:
the current code *has* it

Reply:
LMAO oops

Reply:
okay updated lol